Vagrantfile

```diff
@@ -9,26 +9,31 @@ Vagrant.configure(2) do |config|
   # Every Vagrant development environment requires a box. You can search for
   # boxes at https://atlas.hashicorp.com/search.
 
-  # Create port forward mappings
-  # These are not normally required, since all access happens on the
-  # 192.168.77.77 IP address
-  #
-  # config.vm.network "forwarded_port", guest: 80, host: 80
-  # config.vm.network "forwarded_port", guest: 443, host: 443
-  # config.vm.network "forwarded_port", guest: 4443, host: 4443
-  # config.vm.network "forwarded_port", guest: 8501, host: 8501
-
   vm_memory = ENV.fetch('VM_MEMORY', 10 * 1024).to_i
   vm_cpus = ENV.fetch('VM_CPUS', 4).to_i
 
+  os=`uname`.strip
+  if os == 'Darwin'
+    default_if=`route get default | grep interface`.split(" ").last
+  else
+    default_if=`/sbin/route | grep default`.split(" ").last
+  end
+  # Ugly hack to warn user about not using a host bridge with libvirt
+  user_warned_about_bridge = false
+
+  net_config = {
+    :use_dhcp_assigned_default_route => true
+  }
+
   # Create a private network, which allows host-only access to the machine
   # using a specific IP.
-  config.vm.network "private_network", ip: "192.168.77.77"
 
   config.vm.provider "virtualbox" do |vb, override|
     # Need to shorten the URL for Windows' sake
-    override.vm.box = "https://cf-opensusefs2.s3.amazonaws.com/vagrant/scf-virtualbox-v2.0.3.box"
-
+    override.vm.box = "https://cf-opensusefs2.s3.amazonaws.com/vagrant/scf-virtualbox-v2.0.5.box"
+    net_config[:nic_type] = "virtio"
+    net_config[:bridged] = default_if
+    override.vm.network "public_network", net_config
```

Comment:
There's a lot of branching in the ways the net_config needs to be changed depending on Linux/OSX, bridged/NAT, DHCP/no-DHCP (only relevant for NAT), libvirt/vbox. I think the configuration hash needs to be constructed piecemeal, otherwise we'll have more conditional branches than necessary.